bridge-ext/src/main/java/com/liferay/faces/bridge/ext/renderkit/html_basic/internal/ResourceRendererLiferayImpl.java

```diff
@@ -0,0 +1,233 @@
+import javax.faces.event.ComponentSystemEventListener;
+import javax.faces.event.ListenerFor;
+import javax.faces.event.PostAddToViewEvent;
+import javax.faces.render.RenderKit;
+import javax.faces.render.RenderKitWrapper;
+import javax.faces.render.Renderer;
+import javax.faces.render.RendererWrapper;
+
+import com.liferay.faces.util.logging.Logger;
+import com.liferay.faces.util.logging.LoggerFactory;
+
+
+/**
+ * @author  Kyle Stiemann
+ */
+@ListenerFor(systemEventClass = PostAddToViewEvent.class)
+public class ResourceRendererLiferayImpl extends RendererWrapper implements ComponentSystemEventListener, StateHolder {
+
+	// Logger
+	private static final Logger logger = LoggerFactory.getLogger(ResourceRendererLiferayImpl.class);
+
+	// Private Data Members
+	private String componentFamily;
+	private String rendererType;
+	private boolean renderHeadResourceIds;
+	private boolean transientFlag;
+	private Renderer wrappedRenderer;
+
+	/**
+	 * This zero-arg constructor is required by the {@link javax.faces.component.StateHolderSaver} class during the
+	 * RESTORE_VIEW phase of the JSF lifecycle. The reason why this class is involved in restoring state is because the
+	 * {@link javax.faces.component.UIComponent.ComponentSystemEventListenerAdapter} implements {@link
+	 * javax.faces.component.StateHolder} and will attempt to restore the state of any class in the restored view that
+	 * implements {@link ComponentSystemEventListener}. It does this first by instantiating the class with a zero-arg
+	 * constructor, and then calls the {@link #restoreState(FacesContext, Object)} method.
+	 */
+	public ResourceRendererLiferayImpl() {
+		// Defer initialization of wrappedRenderer until restoreState(FacesContext, Object) is called.
+	}
+
+	public ResourceRendererLiferayImpl(Renderer renderer, String componentFamily, String rendererType,
+		boolean renderHeadResourceIds) {
+
+		this.wrappedRenderer = renderer;
+		this.componentFamily = componentFamily;
+		this.rendererType = rendererType;
+		this.renderHeadResourceIds = renderHeadResourceIds;
+	}
+
+	private static boolean isLiferayFacesBridgeInlineScript(UIComponent componentResource, String resourceId) {
+
+		boolean liferayBridgeInlineScript = false;
+
+		if ((resourceId != null) && resourceId.toLowerCase(Locale.ENGLISH).contains("inlinescript")) {
+
+			Class<? extends UIComponent> clazz = componentResource.getClass();
+			String className = clazz.getName();
+			liferayBridgeInlineScript = className.startsWith("com.liferay.faces.bridge");
+		}
+
+		return liferayBridgeInlineScript;
+	}
+
+	private static boolean isResource(UIComponent componentResource, String resourceName, String resourceLibrary) {
+		return (isScriptResource(resourceName, resourceLibrary) || isStyleSheetResource(resourceName)) &&
+			!isLiferayFacesBridgeInlineScript(componentResource, resourceName);
+	}
+
+	private static boolean isRichFacesReslibResource(String resourceName, String resourceLibrary) {
```

Comment:
@stiemannkj1 Just curious, did you encounter use-cases for `isLiferayFacesBridgeInlineScript()` and `isRichFacesReslibResource()` during testing of your workaround as you developed it? These look more like corner-cases than use-cases that we would typically encounter.

Reply:
@ngriffin7a, most of this code is ripped directly from Bridge Impl where similar detection is needed. The only thing that's changed is how we detect inline scripts since I don't have access to the `InlineScript` class in Bridge Ext. The point of the code is to make sure that we are only adding `id` attributes to `<script>`s with a `src` and `<link>`s with an `href`. The idea is that we only want to detect resources, not inline styles or scripts.

Bridge Impl code: https://github.com/liferay/liferay-faces-bridge-impl/blob/2929639ab47c311ce14438b542dbb7f1c0528e68/bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/RenderKitUtil.java#L63-L81

Note that I did remove the BootsFaces non-JSF resource detection since non-JSF resources (such as those from a CDN etc.) should always have the same uniquely identifiable `src` or `href` even if used in different portlets, modes, states, views etc. (so there's no need to add an `id` attribute).